cmd/completion.go

```diff
@@ -0,0 +1,267 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+
+const completionDesc = `
+Generate autocompletions script for Astro for the specified shell (bash or zsh).
+
+This command can generate shell autocompletions. e.g.
+
+	$ astro completion bash
+
+Can be sourced as such
+
+	$ source <(astro completion bash)
+`
+
+var (
+	completionShells = map[string]func(out io.Writer, cmd *cobra.Command) error{
+		"bash": runCompletionBash,
+		"zsh":  runCompletionZsh,
+	}
+)
+
+func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("shell not specified")
+	}
+	if len(args) > 1 {
+		return errors.New("too many arguments, expected only the shell type")
+	}
+	run, found := completionShells[args[0]]
+	if !found {
+		return errors.Errorf("unsupported shell type %q", args[0])
+	}
+
+	return run(out, cmd)
+}
+
+func runCompletionBash(_ io.Writer, cmd *cobra.Command) error {
+	var buf bytes.Buffer
+	err := cmd.Root().GenBashCompletion(&buf)
+	if err != nil {
+		return fmt.Errorf("error while generating bash completion: %v", err)
+	}
+	// remove the command "completion" from auto-completion
+	code := buf.String()
+	code = strings.Replace(code, `commands+=("completion")`, "", -1)
+
+	fmt.Print(code)
+	return nil
+}
+
+func runCompletionZsh(_ io.Writer, cmd *cobra.Command) error {
+	var buf bytes.Buffer
+
+	// zshHead is the header required to declare zsh completion
+	zshHead := "#compdef astro\n"
```

Comment:
Can we not use `cmd.Root().GenZshCompletion(w)` here?

Reply:
not sure if it works or not ... i checked
helm/kubectl/doctl ... let me try.

Reply:
fixed.